server/src/main/java/org/apache/druid/segment/realtime/appenderator/UnifiedIndexerAppenderatorsManager.java

```diff
@@ -248,6 +289,50 @@ private AppenderatorConfig rewriteAppenderatorConfigMemoryLimits(AppenderatorCon
     return new MemoryParameterOverridingAppenderatorConfig(baseConfig, perWorkerLimit);
   }
 
+  private class DatasourceBundle
+  {
+    private final SinkQuerySegmentWalker walker;
+    private final Map<String, List<Appenderator>> taskAppenderatorMap;
+
+    public DatasourceBundle(
+        String dataSource
+    )
+    {
+      this.taskAppenderatorMap = new HashMap<>();
+
+      VersionedIntervalTimeline<String, Sink> sinkTimeline = new VersionedIntervalTimeline<>(
+          String.CASE_INSENSITIVE_ORDER
```

Comment:
Hmm, I'm not sure what is the proper comparator for the timeline. Raised https://github.com/apache/incubator-druid/issues/8320.